frame/merkle-mountain-range/src/mmr/mmr.rs

```diff
@@ -76,19 +84,27 @@ where
 		Self { mmr: mmr_lib::MMR::new(size, Default::default()), leaves }
 	}
 
-	/// Verify proof of a single leaf.
-	pub fn verify_leaf_proof(
+	/// Verify proof for a set of leaves.
+	/// Note, the leaves should be sorted such that corresponding leaves and leaf indices have
+	/// the same position in both the `leaves` vector and the `leaf_indices` vector contained in the
+	/// [primitives::BatchProof]
+	pub fn verify_leaves_proof(
 		&self,
-		leaf: L,
-		proof: primitives::Proof<<T as Config<I>>::Hash>,
+		leaves: Vec<L>,
+		proof: primitives::BatchProof<<T as Config<I>>::Hash>,
 	) -> Result<bool, Error> {
 		let p = mmr_lib::MerkleProof::<NodeOf<T, I, L>, Hasher<HashingOf<T, I>, L>>::new(
 			self.mmr.mmr_size(),
 			proof.items.into_iter().map(Node::Hash).collect(),
 		);
-		let position = mmr_lib::leaf_index_to_pos(proof.leaf_index);
+		let leaves_positions_and_data = proof
```

Comment:
Could you, please, add a check that number of entries in the `leaves` are matching number of entries in the `proof.leaf_indices`? Someone may call this function without checking this and it may return `true` even if proof not valid - e.g. there are entries in `leaves`, but there are no entries in `proof.leaf_indices` -> `leaves_positions_and_data` is empty -> function will return `true`